apis/core/bootstrap/lib/query.ts

```diff
@@ -6,3 +6,12 @@ import env from '@cube-creator/core/env'
 export function lindasQuery(query: SparqlTemplateResult): NamedNode {
   return $rdf.namedNode(`${env.PUBLIC_QUERY_ENDPOINT}?query=${encodeURIComponent(query.toString())}`)
 }
+
+export function lindasQueryTemplate(query: SparqlTemplateResult, ...variables: string[]) {
+  const url = lindasQuery(query)
+  return variables.reduce(replacePlaceholders, url.value)
+}
+
+function replacePlaceholders(prev: string, variable: string) {
```

Comment:
is there any SPARQL parser library so that we may do the replacement at the query AST level? Or the regex approach is good enough anyway?

Reply:
Yes, there is [sparqljs](https://npm.im/sparqljs) but here the case is not exactly parsing SPARQL

The query goes to the query string, urlencoded. The replacement is to insert template placeholder like `{VAR}` into it. The braces need not be escaped